# SEANCE 9 - Des tests statistiques au plan d’expérience

## Ce que vous savez faire maintenant

À l’issue de ce cours, vous savez :

- analyser des données expérimentales,
- formuler une question statistique claire,
- poser des hypothèses $H_0$ et $H_1$,
- choisir un test adapté (Student, Welch, apparié, corrélation),
- quantifier un effet avec une régression linéaire,
- interpréter une p-value et un intervalle de confiance,
- conclure comme un ingénieur (décision + limites).

Ces outils permettent de répondre à des questions du type :

> *« Ce paramètre a-t-il un effet mesurable ? »*

---

## Limite naturelle des tests vus dans ce cours

Les tests statistiques et la régression simple répondent très bien à des situations où :

- un seul paramètre varie,
- ou deux variables sont étudiées à la fois.

Mais en pratique industrielle, les systèmes réels comportent souvent **plusieurs paramètres simultanément** :

- vitesse **et** température,
- charge **et** matériau,
- avance **et** lubrification,
- pression **et** temps de cycle.

À ce stade, multiplier les tests “un par un” devient :
- long,
- inefficace,
- et incapable de révéler des **interactions** entre paramètres.

---

## Idée clé : quand les tests ne suffisent plus

> **Quand on commence à se demander  
> “et si plusieurs paramètres agissaient en même temps ?”,  
> il ne faut plus seulement analyser les données,  
> il faut concevoir les expériences.**

C’est exactement le rôle du **plan d’expérience (DOE – Design of Experiments)**.

---

## Mini-exemple conceptuel – Pourquoi un plan d’expérience ?

### Problème industriel
On cherche à réduire l’usure d’un outil d’usinage.

### Paramètres suspects
- Facteur A : vitesse de rotation (bas / haut)
- Facteur B : lubrification (absente / présente)

### Réponse mesurée
- Usure de l’outil (mm)

---

### Approche “tests simples”
- comparer deux vitesses → test t
- comparer avec/sans lubrification → test t

❌ Problème :
- impossible de savoir si la lubrification est surtout efficace à haute vitesse,
- interaction A×B invisible.

---

### Approche “plan d’expérience”
On teste **toutes les combinaisons** :

| Essai | Vitesse | Lubrification |
|---|---|---|
| 1 | Bas | Non |
| 2 | Haut | Non |
| 3 | Bas | Oui |
| 4 | Haut | Oui |

Avec **seulement 4 essais**, on peut analyser :
- l’effet de la vitesse,
- l’effet de la lubrification,
- l’interaction vitesse × lubrification.

👉 Cette analyse repose exactement sur les notions vues :
- moyenne,
- régression,
- tests d’hypothèses,
- p-values.

---

## Lien direct avec ce cours

| Outil vu | Question associée |
|---|---|
| Test t | Y a-t-il une différence ? |
| Corrélation | Y a-t-il un lien ? |
| Régression | Quel est l’effet d’un paramètre ? |
| **Plan d’expérience** | Quels paramètres comptent vraiment, et comment interagissent-ils ? |

Le plan d’expérience n’est **pas un nouveau monde** :
c’est l’extension naturelle de la régression à **plusieurs facteurs**.

---

## Message clé à retenir

> **Les tests statistiques analysent les données.  
> Le plan d’expérience permet de concevoir les données.**

En ingénierie :
- on analyse pour comprendre,
- on planifie pour optimiser.

---

## Transition le plan d'expérience

Dans la prochaine séance, nous verrons comment :
- structurer des essais,
- réduire le nombre d’expériences,
- identifier les paramètres réellement influents,
- et préparer une optimisation rationnelle.

👉 Ce sera le rôle du **plan d’expérience**, basé sur les outils que vous maîtrisez maintenant.



## Introduction au Plan d’Expérience (DOE)

Ce notebook introduit la notion de **plan d’expérience (Design of Experiments – DOE)**.
L’objectif n’est pas de maîtriser des méthodes complexes, mais de comprendre :

- pourquoi les tests statistiques simples ne suffisent plus,
- comment **concevoir intelligemment des essais**,
- comment relier DOE, régression et tests d’hypothèses.

---

## 1. Pourquoi un plan d’expérience ?

Dans les notebooks précédents, vous avez appris à :
- tester une moyenne,
- comparer deux procédés,
- détecter un lien,
- quantifier un effet (régression).

Ces outils répondent à des questions du type :

> *« Ce paramètre a-t-il un effet ? »*

Mais en pratique industrielle, on se pose très vite une autre question :

> *« Parmi plusieurs paramètres, lesquels sont réellement influents,
> et comment agissent-ils ensemble ? »*

C’est exactement le rôle du **plan d’expérience**.

---

## 2. Définition simple du plan d’expérience

> Un **plan d’expérience** est une méthode qui consiste à :
> - choisir des paramètres à faire varier (**facteurs**),
> - définir leurs valeurs (**niveaux**),
> - organiser les essais,
> afin d’extraire un **maximum d’information avec un minimum d’expériences**.

Le DOE ne remplace pas les statistiques :
👉 il **les utilise**, mais de manière structurée.

---

## 3. Notions clés à connaître

### Facteur
Paramètre que l’on fait varier volontairement  
(ex : vitesse, température, matériau).

### Niveau
Valeur prise par un facteur  
(ex : bas / haut, 1000 / 2000 tr/min).

### Réponse
Grandeur mesurée  
(ex : usure, rugosité, durée de vie).

---

## 4. Limite de la méthode “un paramètre à la fois”

Faire varier un paramètre à la fois peut sembler logique,
mais cette approche présente deux limites majeures :

- elle nécessite beaucoup d’essais,
- elle ne permet pas de détecter les **interactions**.

> Une interaction signifie que l’effet d’un paramètre dépend
> de la valeur d’un autre paramètre.

---

## 5. Principe du plan factoriel simple (2 facteurs)

Avec deux facteurs A et B à deux niveaux (bas / haut),
un plan factoriel complet consiste à tester **toutes les combinaisons** :

| Essai | Facteur A | Facteur B |
|---|---|---|
| 1 | Bas | Bas |
| 2 | Haut | Bas |
| 3 | Bas | Haut |
| 4 | Haut | Haut |

Avec seulement **4 essais**, on peut analyser :
- l’effet de A,
- l’effet de B,
- l’interaction A × B.

---

# Exercice – Choisir les essais dans un plan d’expérience

## Contexte industriel

Une entreprise d’usinage cherche à **réduire l’usure d’un outil**.

Trois paramètres sont suspectés d’avoir une influence :

- Facteur A : vitesse de rotation  
  (bas / haut)

- Facteur B : avance  
  (faible / élevée)

- Facteur C : lubrification  
  (absente / présente)

La réponse mesurée est :
- l’usure de l’outil (mm).

---

## Question métier

> *Comment organiser les essais pour identifier efficacement
> les paramètres influents, sans multiplier inutilement les expériences ?*

---

## Travail demandé

### 1) Identifier les facteurs et niveaux
Lister clairement :
- les facteurs étudiés,
- leurs niveaux.

---

### 2) Nombre d’essais possibles (raisonnement)
Répondre aux questions suivantes :

- Combien d’essais seraient nécessaires pour tester **toutes les combinaisons**
  des 3 facteurs à 2 niveaux ?
- Pourquoi ce nombre peut-il devenir problématique en industrie ?

*(Aucun calcul complexe n’est attendu, seulement un raisonnement.)*

---

### 3) Choisir un sous-ensemble d’essais

On vous impose une contrainte :
> **Vous ne pouvez réaliser que 4 essais.**

- Proposez **4 essais** (combinaisons de facteurs) que vous jugez pertinents.
- Présentez-les sous forme de tableau.

---

### 4) Justifier votre choix (point clé)

Expliquez :
- pourquoi ces essais ont été choisis,
- quels effets vous espérez pouvoir analyser,
- ce que vous acceptez de **ne pas pouvoir conclure**.

---

### 5) Interprétation ingénieur

Rédiger un court paragraphe répondant à la question suivante :

> *En quoi votre choix d’essais est-il un compromis
> entre coût expérimental et information obtenue ?*

---

## Message clé de l’exercice

> **Un bon plan d’expérience n’est pas celui qui teste tout,
> mais celui qui teste intelligemment.**



## Correction raisonnée – Exercice DOE : choisir les essais

> Il n’existe pas une unique “bonne” réponse : un plan d’expérience est un **compromis**
> entre le coût des essais et l’information visée.  
> La correction ci-dessous donne une démarche attendue et **une proposition cohérente**.

---

<details>
<summary><strong>1) Identifier facteurs et niveaux</strong></summary>

Facteurs (à 2 niveaux) :

- **A : Vitesse de rotation**  
  - Bas (−) / Haut (+)

- **B : Avance**  
  - Faible (−) / Élevée (+)

- **C : Lubrification**  
  - Absente (−) / Présente (+)

Réponse :
- **Usure de l’outil** (mm)

</details>

---

<details>
<summary><strong>2) Combien d’essais pour tester toutes les combinaisons ? Pourquoi c’est un problème ?</strong></summary>

- Avec 3 facteurs à 2 niveaux, un plan factoriel complet nécessite :
$$
2^3 = 8 \text{ essais}
$$

Pourquoi cela peut être coûteux en industrie :
- temps machine mobilisé,
- coût matière/outils,
- disponibilité opérateur/métrologie,
- reproductibilité (il faudrait idéalement des répétitions),
- risque d’évolution de conditions (dérive machine, température atelier…).

Donc, avec des ressources limitées, on cherche un plan **plus économique**.

</details>

---

<details>
<summary><strong>3) Avec seulement 4 essais : que peut-on viser raisonnablement ?</strong></summary>

Avec 4 essais (au lieu de 8), on ne peut pas tout estimer séparément.

Deux objectifs réalistes possibles :

### Option 1 – “Screening” (dépistage)
- identifier rapidement si certains facteurs semblent très influents
- accepter qu’on ne distingue pas bien les interactions

### Option 2 – Comparer deux “recettes” extrêmes
- opposer un réglage “favorable” et “défavorable”
- utile si l’objectif est une décision rapide (et non une modélisation)

La correction propose ci-dessous une option cohérente pour un dépistage.

</details>

---

<details>
<summary><strong>4) Proposition de 4 essais (plan fractionnaire classique)</strong></summary>

Une proposition simple et cohérente est un **fractionnaire 2^(3-1) = 4 essais**.

On choisit les 4 combinaisons suivantes (codage − / +) :

| Essai | A (Vitesse) | B (Avance) | C (Lubrification) |
|---|---|---|---|
| 1 | − | − | − |
| 2 | + | − | + |
| 3 | − | + | + |
| 4 | + | + | − |

Cette sélection est structurée : elle “balance” les niveaux pour éviter un plan trop biaisé
(par exemple, éviter d’avoir toujours C = + quand A = +, etc.).

</details>

---

<details>
<summary><strong>5) Justification (ce que ce plan permet / ne permet pas)</strong></summary>

### Ce que ce plan permet (objectif minimal)
- Comparer l’usure entre des configurations très différentes.
- Avoir une première idée des effets “principaux” :
  - vitesse, avance, lubrification

### Mais attention : limite fondamentale (aliasing)
Avec 4 essais et 3 facteurs, on a une contrainte :

- les effets principaux sont **confondu(e)s** avec certaines interactions.
- concrètement, on ne peut pas séparer de façon sûre :
  - l’effet de A **et** l’interaction B×C,
  - l’effet de B **et** l’interaction A×C,
  - l’effet de C **et** l’interaction A×B.

Donc :
> Si on observe un effet important sur A, il peut provenir de A ou d’une interaction associée.

### Interprétation ingénieur attendue
Ce plan est adapté si :
- on veut un dépistage rapide,
- on est prêt à faire une deuxième phase si un effet apparaît.

</details>

---

<details>
<summary><strong>6) Exemple de conclusion “ingénieur” attendue</strong></summary>

> “Avec seulement 4 essais, nous proposons un plan fractionnaire structuré permettant
> de comparer des configurations contrastées et d’obtenir une première indication des paramètres influents.
> Ce plan est un compromis coût/information : il peut révéler des tendances fortes,
> mais certaines conclusions restent ambiguës car des interactions sont confondues avec les effets principaux.
> Si un facteur semble dominant, une seconde phase d’essais sera nécessaire pour confirmer
> et distinguer effet principal et interactions.”

</details>

---

<details>
<summary><strong>7) Ce qu’on attend comme réflexion finale (coût vs info)</strong></summary>

Un bon raisonnement doit mentionner :

- **coût réduit** (4 essais au lieu de 8, sans même parler de répétitions)
- **information partielle** :
  - on peut dépister,
  - mais pas modéliser finement
- **décision** :
  - soit on accepte le compromis pour une décision rapide,
  - soit on planifie une phase 2 (plan complet ou essais ciblés)

</details>


In [ ]:
import numpy as np
import pandas as pd
import statsmodels.api as sm

# ------------------------------------------------------------
# Plan fractionnaire 2^(3-1) : 4 essais
# Codage : -1 = niveau bas, +1 = niveau haut
# Essais proposés :
# 1: A- B- C-
# 2: A+ B- C+
# 3: A- B+ C+
# 4: A+ B+ C-
# ------------------------------------------------------------
design = pd.DataFrame({
    "A_vitesse": [-1, +1, -1, +1],
    "B_avance":  [-1, -1, +1, +1],
    "C_lubr":    [-1, +1, +1, -1],
})

# Pour affichage "lisible" (optionnel)
design_readable = design.replace({-1: "Bas", +1: "Haut"})
design_readable.index = [f"Essai {i}" for i in range(1, 5)]
display(design_readable)

# ------------------------------------------------------------
# Simulation d'une réponse (usure en mm) pour l'exemple
# IMPORTANT : c'est une simulation pédagogique, pas une vérité industrielle
# y = base + effetA*A + effetB*B + effetC*C + bruit
# ------------------------------------------------------------
rng = np.random.default_rng(0)
base = 0.30
effA = 0.05   # ex : haute vitesse -> +usure
effB = 0.03   # ex : forte avance  -> +usure
effC = -0.04  # ex : lubrification présente -> -usure
sigma = 0.005 # petit bruit (mais avec 4 essais, ça ne "sauve" pas le ddl)

y = (base
     + effA*design["A_vitesse"]
     + effB*design["B_avance"]
     + effC*design["C_lubr"]
     + rng.normal(0, sigma, size=len(design)))

design["Usure_mm"] = y
display(design)

# ------------------------------------------------------------
# Ajustement régression sur effets principaux (A, B, C)
# ATTENTION : avec 4 essais et (intercept + 3 effets) = 4 paramètres
# -> modèle saturé : ajustement parfait, ddl résiduel = 0
# -> p-values et tests classiques ne sont pas interprétables
# ------------------------------------------------------------
X = sm.add_constant(design[["A_vitesse", "B_avance", "C_lubr"]])
model = sm.OLS(design["Usure_mm"], X).fit
print(model.summary())

# Affiche uniquement les coefficients (OK même si modèle saturé)
print("Coefficients estimés (modèle saturé, pas de p-values exploitables) :")
print(model.params)

print("\n⚠️ Note : avec 4 essais et 4 paramètres (constante + A + B + C),")
print("ddl résiduels = 0 → impossible d'estimer l'erreur → pas de tests/p-values fiables.")


print("\n⚠️ Remarque pédagogique :")
print("- Ici, 4 essais = modèle saturé (ddl résiduel = 0).")
print("- On peut estimer des effets, mais on ne peut pas estimer correctement l'erreur.")
print("- En pratique, on ajoute des répétitions (réplicats) ou des points centre pour faire de l'inférence.")


A_vitesse B_avance C_lubr
Essai 1       Bas      Bas    Bas
Essai 2      Haut      Bas   Haut
Essai 3       Bas     Haut   Haut
Essai 4      Haut     Haut    Bas

A_vitesse  B_avance  C_lubr  Usure_mm
0         -1        -1      -1  0.260629
1          1        -1       1  0.279339
2         -1         1       1  0.243202
3          1         1      -1  0.420525

Coefficients estimés (modèle saturé, pas de p-values exploitables) :
const        0.300924
A_vitesse    0.049008
B_avance     0.030940
C_lubr      -0.039653
dtype: float64

⚠️ Note : avec 4 essais et 4 paramètres (constante + A + B + C),
ddl résiduels = 0 → impossible d'estimer l'erreur → pas de tests/p-values fiables.

⚠️ Remarque pédagogique :
- Ici, 4 essais = modèle saturé (ddl résiduel = 0).
- On peut estimer des effets, mais on ne peut pas estimer correctement l'erreur.
- En pratique, on ajoute des répétitions (réplicats) ou des points centre pour faire de l'inférence.


## Interprétation détaillée des résultats du plan d’expérience

Les coefficients ci-dessous proviennent de l’ajustement d’un modèle linéaire
sur un **plan d’expérience fractionnaire** à trois facteurs, codés selon la convention
$-1 / +1$ (niveau bas / niveau haut).

Le modèle utilisé est :
$$
Y = \beta_0 + \beta_A A + \beta_B B + \beta_C C
$$
où $Y$ représente l’usure de l’outil (en mm).

---

### 1. Interprétation de la constante (intercept)

**Constante :**
$$
\beta_0 = 0.300924
$$

Cette valeur correspond à l’usure **prédite au centre du domaine expérimental**,
c’est-à-dire pour :
$$
A = 0,\quad B = 0,\quad C = 0
$$

Même si ces niveaux n’existent pas physiquement, la constante représente :
- une **valeur moyenne de référence**,
- un point de comparaison pour interpréter les effets des facteurs.

---

### 2. Interprétation des coefficients des facteurs

Les facteurs étant codés en $-1 / +1$, chaque coefficient représente
**la moitié de l’effet total** du facteur.

Pour obtenir l’effet réel entre le niveau bas et le niveau haut,
il faut multiplier le coefficient par 2.

---

#### Effet de la vitesse (facteur A)

**Coefficient :**
$$
\beta_A = 0.049008
$$

- Le signe positif indique qu’une **augmentation de la vitesse**
  entraîne une **augmentation de l’usure**.
- L’effet total entre vitesse basse et vitesse haute est :
$$
\Delta Y_A = 2 \times 0.049008 \approx 0.098 \text{ mm}
$$

**Interprétation ingénieur :**
La vitesse est un facteur fortement influent.
Passer à une vitesse élevée augmente significativement l’usure de l’outil.

---

#### Effet de l’avance (facteur B)

**Coefficient :**
$$
\beta_B = 0.030940
$$

- Le signe positif indique qu’une **avance élevée**
  augmente l’usure.
- Effet total :
$$
\Delta Y_B = 2 \times 0.030940 \approx 0.062 \text{ mm}
$$

**Interprétation ingénieur :**
L’avance a un effet réel mais plus modéré que la vitesse.
C’est un levier d’optimisation secondaire.

---

#### Effet de la lubrification (facteur C)

**Coefficient :**
$$
\beta_C = -0.039653
$$

- Le signe négatif indique que la **présence de lubrification**
  réduit l’usure.
- Effet total :
$$
\Delta Y_C = 2 \times (-0.039653) \approx -0.079 \text{ mm}
$$

**Interprétation ingénieur :**
La lubrification a un effet bénéfique marqué,
réduisant l’usure de manière comparable à l’effet de la vitesse,
mais en sens opposé.

---

### 3. Comparaison des effets

| Facteur | Effet total estimé (mm) | Importance relative |
|-------|-------------------------|---------------------|
| Vitesse (A) | +0.098 | Très fort |
| Lubrification (C) | −0.079 | Très fort |
| Avance (B) | +0.062 | Modéré |

Les paramètres les plus influents sont donc :
1. la vitesse,
2. la lubrification,
3. l’avance.

---

### 4. Limite d’interprétation (plan fractionnaire)

Ce plan d’expérience comporte seulement 4 essais et ne permet pas de séparer
les **effets principaux** des **interactions** entre facteurs.

En particulier :
- l’effet de la vitesse est confondu avec l’interaction avance × lubrification,
- l’effet de l’avance est confondu avec l’interaction vitesse × lubrification,
- l’effet de la lubrification est confondu avec l’interaction vitesse × avance.

Les effets estimés doivent donc être interprétés comme des **effets apparents**.

---

### 5. Conclusion d’ingénieur

Le plan fractionnaire met en évidence des effets importants associés
à la vitesse et à la lubrification.
Ces paramètres doivent être considérés en priorité pour réduire l’usure.

Cependant, en raison des interactions confondues et de l’absence de répétitions,
une seconde phase d’essais est nécessaire pour confirmer ces résultats
et isoler les mécanismes en jeu.

---

### Message clé à retenir

> **Dans un plan d’expérience codé $-1 / +1$,  
> les coefficients sont des demi-effets.  
> Le plan fractionnaire permet de dépister, pas de conclure définitivement.**
